bolt-sidecar/src/state/execution.rs

```diff
@@ -609,7 +617,39 @@ mod tests {
         test_util::{create_signed_inclusion_request, default_test_transaction, launch_anvil},
     };
 
-    use super::*;
+    #[test]
+    fn test_compute_diff_no_templates() {
+        let block_templates = HashMap::new();
+        let sender = Address::random();
+
+        let (nonce_diff, balance_diff, highest_slot) = compute_diffs(&block_templates, &sender);
+
+        assert_eq!(nonce_diff, 0);
+        assert_eq!(balance_diff, U256::ZERO);
+        assert_eq!(highest_slot, 0);
+    }
+
+    #[test]
+    fn test_compute_diff_single_template() {
+        let mut block_templates = HashMap::new();
+        let sender = Address::random();
+
+        let mut diffs = HashMap::new();
+        diffs.insert(sender.clone(), (1, U256::from(2)));
+
+        let mut state_diff = StateDiff::default();
+        state_diff.diffs = diffs.clone();
+
+        let block_template = BlockTemplate { state_diff, signed_constraints_list: vec![] };
+
+        block_templates.insert(10, block_template);
+
+        let (nonce_diff, balance_diff, highest_slot) = compute_diffs(&block_templates, &sender);
+
+        assert_eq!(nonce_diff, 1);
+        assert_eq!(balance_diff, U256::from(2));
+        assert_eq!(highest_slot, 10);
```

Comment:
Can you add comments for the steps, so that I can read the test without simulating the code in my head?

Reply:
Can't run a RISCV VM in your head in 2024? ngmi

Reply:
better now?